### DISCUSSION TITLE
scan Container Image in CycloneDX does not provided version in report

### DISCUSSION THREAD

**Discussion by gbloquel** [aquasecurity/trivy#7152] [Bugs]
### Description

When I produce a report in CycloneDX for a container Image I expected to have a version in the component Root and the component Root Name contains only the imageName

The command applied is `trivy image --format cyclonedx nginx:1.19`

If you confirm that is a bug, i can provided a PR.

### Desired Behavior

Expected 

 `metadata.component.name=nginx`
 `metadata.component.version=1.19`

<pre>
{
  "$schema": "http://cyclonedx.org/schema/bom-1.6.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.6",
  "serialNumber": "urn:uuid:ff279ebd-a258-45dc-a4f3-a933759bec41",
  "version": 1,
  "metadata": {
    "timestamp": "2024-07-12T09:26:52+00:00",
    "tools": {
      "components": [
        {
          "type": "application",
          "group": "aquasecurity",
          "name": "trivy",
          "version": "dev"
        }
      ]
    },
    "component": {
      "bom-ref": "pkg:oci/nginx@sha256%3Adf13abe416e37eb3db4722840dd479b00ba193ac6606e7902331dcea50f4f1f2?arch=amd64&repository_url=index.docker.io%2Flibrary%2Fnginx",
      "type": "container",
      "name": "nginx",
      "version": "1.19",
      "purl": "pkg:oci/nginx@sha256%3Adf13abe416e37eb3db4722840dd479b00ba193ac6606e7902331dcea50f4f1f2?arch=amd64&repository_url=index.docker.io%2Flibrary%2Fnginx",

</pre>

### Actual Behavior

Actual the Name contains Image Name and Version.
`metadata.component.name=dockerImage:version`
here `metadata.component.name=nginx:1.19`

<pre>
{
  "$schema": "http://cyclonedx.org/schema/bom-1.6.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.6",
  "serialNumber": "urn:uuid:f6fc780e-95dd-4e39-8bdb-d8d3f64d502b",
  "version": 1,
  "metadata": {
    "timestamp": "2024-07-12T09:46:07+00:00",
    "tools": {
      "components": [
        {
          "type": "application",
          "group": "aquasecurity",
          "name": "trivy",
          "version": "0.53.0"
        }
      ]
    },
    "component": {
      "bom-ref": "pkg:oci/nginx@sha256%3Adf13abe416e37eb3db4722840dd479b00ba193ac6606e7902331dcea50f4f1f2?arch=amd64&repository_url=index.docker.io%2Flibrary%2Fnginx",
      "type": "container",
      "name": "nginx:1.19",
      "purl": "pkg:oci/nginx@sha256%3Adf13abe416e37eb3db4722840dd479b00ba193ac6606e7902331dcea50f4f1f2?arch=amd64&repository_url=index.docker.io%2Flibrary%2Fnginx",
      "properties": [

</pre>

### Reproduction Steps

```bash
1. The command applied is `trivy image --format cyclonedx nginx:1.19`
```

### Target

Container Image

### Scanner

None

### Output Format

CycloneDX

### Mode

None

### Debug Output

```bash
trivy image --format cyclonedx nginx:1.19 --debug
<pre>
2024-07-12T11:55:12+02:00	DEBUG	Cache dir	dir="/home/personal/.cache/trivy"
2024-07-12T11:55:12+02:00	DEBUG	Parsed severities	severities=[UNKNOWN LOW MEDIUM HIGH CRITICAL]
2024-07-12T11:55:12+02:00	DEBUG	Ignore statuses	statuses=[]
2024-07-12T11:55:12+02:00	INFO	"--format cyclonedx" disables security scanning. Specify "--scanners vuln" explicitly if you want to include vulnerabilities in the CycloneDX report.
2024-07-12T11:55:12+02:00	DEBUG	Enabling misconfiguration scanners	scanners=[azure-arm cloudformation dockerfile helm kubernetes terraform terraformplan-json terraformplan-snapshot]
2024-07-12T11:55:12+02:00	DEBUG	Initializing scan cache...	type="fs"
2024-07-12T11:55:12+02:00	DEBUG	[nuget] The nuget packages directory couldn't be found. License search disabled
2024-07-12T11:55:12+02:00	DEBUG	[image] Detected image ID	image_id="sha256:f0b8a9a541369db503ff3b9d4fa6de561b300f7363920c2bff4577c6c24c5cf6"
2024-07-12T11:55:12+02:00	DEBUG	[image] Detected diff ID	diff_ids=[sha256:02c055ef67f5904019f43a41ea5f099996d8e7633749b6e606c400526b2c4b33 sha256:431f409d4c5a8f79640000705665407ff22d73e043472cb1521faa6d83afc5e8 sha256:4b8db2d7f35aa38ac283036f2c7a453ebfdcc8d7e83a2bf3b55bf8847f8fafaf sha256:c9732df61184e9e8d08f96c6966190c59f507d8f57ea057a4610f145c59e9bc4 sha256:eeb14ff930d4c2c04ece429112c16a536985f0cba6b13fdb52b00853107ab9c4 sha256:f0f30197ccf95e395bbf4efd65ec94b9219516ae5cafe989df4cf220eb1d6dfa]
2024-07-12T11:55:12+02:00	DEBUG	[image] Detected base layers	diff_ids=[sha256:02c055ef67f5904019f43a41ea5f099996d8e7633749b6e606c400526b2c4b33]
2024-07-12T11:55:12+02:00	INFO	Detected OS	family="debian" version="10.9"
2024-07-12T11:55:12+02:00	INFO	Number of language-specific files	num=0
{
  "$schema": "http://cyclonedx.org/schema/bom-1.6.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.6",
  "serialNumber": "urn:uuid:41279946-9d7e-4652-96fa-49e2da8becd4",
  "version": 1,
  "metadata": {
    "timestamp": "2024-07-12T09:55:12+00:00",
    "tools": {
      "components": [
        {
          "type": "application",
          "group": "aquasecurity",
          "name": "trivy",
          "version": "0.53.0"
        }
      ]
    },
    "component": {
      "bom-ref": "pkg:oci/nginx@sha256%3Adf13abe416e37eb3db4722840dd479b00ba193ac6606e7902331dcea50f4f1f2?arch=amd64&repository_url=index.docker.io%2Flibrary%2Fnginx",
      "type": "container",
      "name": "nginx:1.19",
...
</pre>
```

### Operating System

Ubuntu 24.04

### Version

```bash
$ trivy --version
Version: 0.53.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-07-11 12:12:28.513917276 +0000 UTC
  NextUpdate: 2024-07-11 18:12:28.513916895 +0000 UTC
  DownloadedAt: 2024-07-11 14:22:15.746626953 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2024-07-11 01:04:59.124718973 +0000 UTC
  NextUpdate: 2024-07-14 01:04:59.124718802 +0000 UTC
  DownloadedAt: 2024-07-11 14:23:58.083131297 +0000 UTC
```

### Checklist

- [ ] Run `trivy clean --all`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @gbloquel 
Thanks for your report!

@knqyf263 wdyt? Can we say that `tag` is [version](https://cyclonedx.org/docs/1.6/json/#metadata_component_version) of image/component?

**Reply by knqyf263** [reply]
I don't think so. PURL doesn't use a tag for the version of [the oci type](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#oci). They use a digest as a version. It makes more sense to add a digest in the version field to comply with PURL.

**Reply by gbloquel** [reply]
In the link [version](https://cyclonedx.org/docs/1.6/json/#metadata_component_version) referenced by @DmitriyLewen we can read  `The version should ideally comply with semantic versioning but is not enforced.`
For me a semantic version is  a version number with MAJOR.MINOR.PATCH.   That's why I thought the version made sense.

I understand your remark but you are talking about `PURL` which is referenced elsewhere in the SBOM document.

**Reply by knqyf263** [reply]
> The version should ideally comply with semantic versioning but is not enforced.

An image tag is almost free text. It doesn't comply with semver at all. In your case, the image uses the version by chance.

>I understand your remark but you are talking about PURL which is referenced elsewhere in the SBOM document.

Since SBOM needs some package identifier and Trivy uses PURL, we are trying to get aligned with PURL.
